packages/moonstone/TooltipDecorator/util.js

```diff
@@ -75,19 +79,26 @@ const adjustDirection = function (tooltipDirection, overflow, rtl) {
  * @private
  */
 const calcOverflow = function (tooltipNode, clientNode, tooltipDirection, tooltipHeight) {
+	const rightDelta = tooltipNode.width > clientNode.left + (clientNode.width / 4) * 3;
+	const leftDelta = tooltipNode.width > window.innerWidth - clientNode.right - (clientNode.width / 3) * 2;
+	const isTooltipWide = (tooltipNode.width > window.innerWidth) ||
+		(leftDelta && rightDelta);
+
 	if (tooltipDirection === 'above' || tooltipDirection === 'below') {
 		return {
 			isOverTop: clientNode.top - tooltipNode.height - tooltipHeight < 0,
 			isOverBottom: clientNode.bottom + tooltipNode.height + tooltipHeight > window.innerHeight,
 			isOverLeft: clientNode.left - tooltipNode.width + clientNode.width / 2 < 0,
-			isOverRight: clientNode.right + tooltipNode.width - clientNode.width / 2 > window.innerWidth
+			isOverRight: clientNode.right + tooltipNode.width - clientNode.width / 2 > window.innerWidth,
```

Comment:
I've never really understood why these lines use this math on several parts when a check to see if something overflows the right would be `tooltipNode.getBoundingClientRect().right > window.innerWidth` or more directly in this context, since we already have the `getBoundingClientRect` object: `tooltipNode.right > window.innerWidth`. Perhaps it's because the tooltip isn't in that position yet, so `right` isn't accurate at the time of this execution.

Reply:
It is because we want to start the tooltip positioning in the middle of the client node. Which is why we need to add the extra calculation with `clientNode.width / 2` and account for that positioning difference.